src/types/index.ts

```diff
@@ -0,0 +1,33 @@
+export interface FormData {
+  id: string;
+  name: string;
+  age: number;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  gender: string;
+  acceptTerms: boolean;
+  profilePicture: string | null;
+  country: string;
+  createdAt: string;
+  formType: 'uncontrolled' | 'controlled';
+}
+
+export interface Country {
+  code: string;
+  name: string;
+}
+
+export type ModalType = 'uncontrolled' | 'controlled' | null;
```

Comment:
You can reuse this type in [src/App.tsx ](https://github.com/novogran/ReactForms/blob/8a7d75a87a70867cbfaa56facf0fcd70c723dc74/src/App.tsx#L13)